packages/framework/src/services/kafka/kafka-broker.client.ts

```diff
@@ -0,0 +1,48 @@
+import { randomUUID } from 'crypto';
+import { Consumer, Kafka, Message, Producer } from 'kafkajs';
+
+import { BrokerProxy } from '../../models/interfaces/broker-proxy.interface';
+
+interface KafkaClients {
+  producer: Producer;
+  consumer: Consumer;
+}
+
+interface KafkaProducePayload {
+  topic: string;
+  messages: string[];
+}
+
+export interface KafkaConsumePayload {
+  topic: string;
+  message: Buffer | null;
+}
+
+export class KafkaBrokerClient extends BrokerProxy<unknown> {
+  #client: KafkaClients;
+
+  async connect(): Promise<void> {
+    const kafka = new Kafka({ clientId: randomUUID(), brokers: [this.broker] });
+    this.#client = {
+      producer: kafka.producer(),
+      consumer: kafka.consumer(),
+    };
+
+    await this.#client.producer.connect();
+    await this.#client.consumer.connect();
+  }
+
+  async publish(payload: KafkaProducePayload): Promise<void> {
+    const messages: Message[] = payload.messages.map((message) => ({ value: message }));
+    await this.#client.producer.send({ topic: payload.topic, messages });
```

Comment:
Could it be helpful to return what `send` returns?

Reply:
The return payload is a bit ambiguous, we can keep the implementation as it is until we fully understand the return metadata.